src/Features/LanguageServer/Protocol/Handler/Breakpoints/ValidateBreakableRangeHandler.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.Debugging;
+
+namespace Microsoft.VisualStudio.LanguageServer.Protocol
+{
+    [DataContract]
+    internal sealed class ValidateBreakableRangeParams
+    {
+        [DataMember(Name = "textDocument")]
+        public TextDocumentIdentifier TextDocument { get; }
+
+        [DataMember(Name = "range")]
+        public Range Range { get; }
+
+        public ValidateBreakableRangeParams(TextDocumentIdentifier textDocument, Range range)
+        {
+            TextDocument = textDocument;
+            Range = range;
+        }
+    }
+}
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    [ExportRoslynLanguagesLspRequestHandlerProvider, Shared]
+    [ProvidesMethod("textDocument/validateBreakableRange")]
+    internal sealed class ValidateBreakableRangeHandler : AbstractStatelessRequestHandler<LSP.ValidateBreakableRangeParams, LSP.Range?>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ValidateBreakableRangeHandler()
+        {
+        }
+
+        public override string Method => "textDocument/validateBreakableRange";
+
+        public override bool MutatesSolutionState => false;
+        public override bool RequiresLSPSolution => true;
+
+        public override LSP.TextDocumentIdentifier? GetTextDocumentIdentifier(LSP.ValidateBreakableRangeParams request)
+            => request.TextDocument;
+
+        public override async Task<LSP.Range?> HandleRequestAsync(LSP.ValidateBreakableRangeParams request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.Document;
+            if (document == null)
```

Comment:
Potentially should throw here if we weren't able to get a document for the request

Reply:
Done. Why is context.Document nullable if we have to have it?

Reply:
Not all requests provide a document, for example workspace/symbol or workspace diagnostics

Reply:
To expand on this, `RequiresLSPSolution` being `true` and  and `GetTextDocumentIdentifier` not returning null is a contract that says `context.Document` can't be null. It's just annoying to express that in the API :)